Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean

```diff
@@ -248,6 +262,49 @@ theorem hasDerivAt_ofReal_cpow {x : ℝ} (hx : x ≠ 0) {r : ℂ} (hr : r ≠ -1
     · exact hasDerivAt_id ((-x : ℝ) : ℂ)
     · simp [hx]
 
+/-- An alternate statement of `hasDerivAt_ofReal_cpow`. -/
+theorem hasDerivAt_ofReal_cpow' {x : ℝ} (hx : x ≠ 0) {r : ℂ} (hr : r ≠ 0) :
+    HasDerivAt (fun y : ℝ => (y : ℂ) ^ r) (r * x ^ (r - 1)) x := by
+  have := HasDerivAt.const_mul r <| hasDerivAt_ofReal_cpow hx (by rwa [ne_eq, sub_eq_neg_self])
+  simp_rw [sub_add_cancel, mul_div_cancel₀ _ hr] at this
+  exact this
+
+/-- A version of `DifferentiableAt.cpow_const` for a real function. -/
+theorem DifferentiableAt.cpow_const' {f : ℝ → ℝ} {x : ℝ} (hf : DifferentiableAt ℝ f x)
+    (h0 : f x ≠ 0) (h1 : c ≠ 0) :
+    DifferentiableAt ℝ (fun (y : ℝ) => (f y : ℂ) ^ c) x :=
+  (hasDerivAt_ofReal_cpow' h0 h1).differentiableAt.comp x hf
+
+theorem Complex.deriv_cpow_const (hx : x ∈ Complex.slitPlane) :
+    deriv (fun (x : ℂ) ↦ x ^ c) x = c * x ^ (c - 1) :=
+  (hasStrictDerivAt_cpow_const hx).hasDerivAt.deriv
+
+/-- A version of `Complex.deriv_cpow_const` for a real variable. -/
+theorem Complex.deriv_cpow_const' {x : ℝ} (hx : x ≠ 0) (hc : c ≠ 0) :
+    deriv (fun x : ℝ ↦ (x : ℂ) ^ c) x = c * x ^ (c - 1) :=
+  (hasDerivAt_ofReal_cpow' hx hc).deriv
+
+theorem deriv_cpow_const (hf : DifferentiableAt ℂ f x) (hx : f x ∈ Complex.slitPlane) :
```

Comment:
I'm not sure about having `Complex.deriv_cpow_const` and `deriv_cpow_const` when they're both equally about `Complex`

Reply:
I quite agree but I was trying to stick to the naming pattern of the `real` case : 

[Real.deriv_rpow_const](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/Pow/Deriv.html#Real.deriv_rpow_const)

[deriv_rpow_const](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/Pow/Deriv.html#deriv_rpow_const)

Reply:
In any case, thanks for the review!